cedar-policy-validator/src/validation_result.rs

```diff
@@ -77,48 +77,54 @@ impl ValidationResult {
 /// and provides details specific to that kind of problem. The error also records
 /// where the problem was encountered.
 #[derive(Clone, Debug, Error, Eq, PartialEq)]
-#[error("for policy `{}`, {}", location.policy_id(), kind)]
+#[error("for policy `{policy_id}`, {kind}")]
 pub struct ValidationError {
-    location: SourceLocation,
-    kind: ValidationErrorKind,
+    pub(crate) policy_id: PolicyID,
+    pub(crate) source_loc: Option<Loc>,
+    pub(crate) kind: ValidationErrorKind,
 }
 
 impl ValidationError {
-    pub(crate) fn with_policy_id(id: PolicyID, source_loc: Option<Loc>, err: TypeError) -> Self {
+    pub(crate) fn with_policy_id(
```

Comment:
Maybe worth a doc comment on this function pointing out that if `err` has a source loc associated, that overrides the `source_loc` explicitly passed to this function.  Callers might reasonably assume the opposite.